compiler/hie-core/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -116,12 +151,19 @@ runLanguageServer options userHandlers getIdeState = do
 -- | Things that get sent to us, but we don't deal with.
 --   Set them to avoid a warning in VS Code output.
 setHandlersIgnore :: PartialHandlers
-setHandlersIgnore = PartialHandlers $ \_ x -> return x
-    {LSP.cancelNotificationHandler = none
-    ,LSP.initializedHandler = none
+setHandlersIgnore = PartialHandlers $ \WithMessage{..} x -> return x
+    {LSP.initializedHandler = none
     }
     where none = Just $ const $ return ()
 
+cancelHandler :: (LspId -> IO ()) -> PartialHandlers
+cancelHandler cancelRequest = PartialHandlers $ \_ x -> return x
+    {LSP.cancelNotificationHandler = Just $ \msg@NotificationMessage {_params} -> do
+            let reqId = case _params of CancelParams{_id} -> _id
+            cancelRequest reqId
+            whenJust (LSP.cancelNotificationHandler x) ($msg)
```

Comment:
```suggestion
            whenJust (LSP.cancelNotificationHandler x) ($ msg)
```

Otherwise you get ambiguities with TH (and I mean humans rather than just compilers)

Reply:
Good point, I also get `hlint` suggesting to me that I create a type error:
```
hie-core/src/Development/IDE/LSP/LanguageServer.hs:164:56: Warning: Redundant bracket
Found:
  ($msg)
Perhaps:
  $msg
```

Reply:
Yes, fixed in newer HLint! But also a sign of the TH confusion.